File: BKLastDanceFE/src/services/epicService.ts

import { api } from "./api";
import {Epic, Sprint} from "@/types";

export const epicService = {
  getAllEpics: async (): Promise<Epic[]> => {
    const response = await api.get("/api/epics");
    return response.data;
  },
  getEpicsByProject: async (projectId: number): Promise<Epic[]> => {
    const response = await api.get(`/api/epics/project/${projectId}`);
    return response.data;
  },
  getEpicById: async (id: number): Promise<Epic> => {
    const response = await api.get(`/api/epics/${id}`);
    return response.data;
  },
  
  createEpic: async (epic: Omit<Epic, "id">): Promise<Epic> => {
    const response = await api.post("/api/epics", epic);
    return response.data;
  },
  
  updateEpic: async (id: string, epic: Partial<Epic>): Promise<Epic> => {
    const response = await api.put(`/api/epics/${id}`, epic);
    return response.data;
  },
  
  deleteEpic: async (id: string): Promise<void> => {
    await api.delete(`/api/epics/${id}`);
  },
  
  getEpicDependencies: async (epicId: number): Promise<string[]> => {
    const response = await api.get(`/api/epics/${epicId}/dependencies`);
    return response.data;
  },

  
  getEpicEffort: async (epicId: string): Promise<any> => {
    const response = await api.get(`/api/epics/${epicId}/effort`);
    return response.data;
  }
};

File: BKLastDance/src/main/java/com/thesis/projectmanagement/service/EpicService.java
package com.thesis.projectmanagement.service;

import com.thesis.projectmanagement.dto.epic.EpicResponse;
import com.thesis.projectmanagement.dto.epic.EpicRequest;

import java.util.List;

public interface EpicService {
    List<EpicResponse> getAllEpics();
    List<EpicResponse> getEpicsByProjectId(Long projectId);
    List<String> getDependencyTitles(Long epicId);
    EpicResponse getEpicById(Long id);
    EpicResponse createEpic(EpicRequest request);
    EpicResponse updateEpic(Long id, EpicRequest request);
    void cancelEpic(Long id);
}
